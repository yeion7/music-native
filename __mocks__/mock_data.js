export const status = {
  showPlayer: false,
  expand: false,
  isLoading: false,
  isPlaying: false,
  position: null,
  duration: null
};

export const song = {
  id: "0dMDz3n9qLfRXBNlRhK5U7",
  name: "Joe Tex, These Taming Blues",
  duration_ms: 330813,
  preview_url: "https://p.scdn.co/mp3-preview/45d2ad042bf08d7951076097a1f7f15677ab7a89?cid=null",
  artists: [
    {
      external_urls: {
        spotify: "https://open.spotify.com/artist/57kIMCLPgkzQlXjblX7XXP"
      },
      href: "https://api.spotify.com/v1/artists/57kIMCLPgkzQlXjblX7XXP",
      id: "57kIMCLPgkzQlXjblX7XXP",
      name: "Phosphorescent",
      type: "artist",
      uri: "spotify:artist:57kIMCLPgkzQlXjblX7XXP"
    }
  ],
  album: {
    album_type: "album",
    artists: [
      {
        external_urls: {
          spotify: "https://open.spotify.com/artist/57kIMCLPgkzQlXjblX7XXP"
        },
        href: "https://api.spotify.com/v1/artists/57kIMCLPgkzQlXjblX7XXP",
        id: "57kIMCLPgkzQlXjblX7XXP",
        name: "Phosphorescent",
        type: "artist",
        uri: "spotify:artist:57kIMCLPgkzQlXjblX7XXP"
      }
    ],
    external_urls: {
      spotify: "https://open.spotify.com/album/0TticAuqNj26EhgT2Em3qD"
    },
    href: "https://api.spotify.com/v1/albums/0TticAuqNj26EhgT2Em3qD",
    id: "0TticAuqNj26EhgT2Em3qD",
    images: [
      {
        height: 640,
        url: "https://i.scdn.co/image/ccd62a1aa5c66d581622a1be2c86cfd0299a3b45",
        width: 640
      },
      {
        height: 300,
        url: "https://i.scdn.co/image/cc9f4c667dc6fa65839c121262a7e7ba5dd49547",
        width: 300
      },
      {
        height: 64,
        url: "https://i.scdn.co/image/1ea5645d08a452f50c9720aad99d4ea6e8cdd8b6",
        width: 64
      }
    ],
    name: "Aw Come Aw Wry",
    type: "album",
    uri: "spotify:album:0TticAuqNj26EhgT2Em3qD"
  },
  track_number: 3
};

export const playlist = [
  {
    id: "0dMDz3n9qLfRXBNlRhK5U7",
    name: "Joe Tex, These Taming Blues",
    duration_ms: 330813,
    preview_url: "https://p.scdn.co/mp3-preview/45d2ad042bf08d7951076097a1f7f15677ab7a89?cid=null",
    artists: [
      {
        external_urls: {
          spotify: "https://open.spotify.com/artist/57kIMCLPgkzQlXjblX7XXP"
        },
        href: "https://api.spotify.com/v1/artists/57kIMCLPgkzQlXjblX7XXP",
        id: "57kIMCLPgkzQlXjblX7XXP",
        name: "Phosphorescent",
        type: "artist",
        uri: "spotify:artist:57kIMCLPgkzQlXjblX7XXP"
      }
    ],
    album: {
      album_type: "album",
      artists: [
        {
          external_urls: {
            spotify: "https://open.spotify.com/artist/57kIMCLPgkzQlXjblX7XXP"
          },
          href: "https://api.spotify.com/v1/artists/57kIMCLPgkzQlXjblX7XXP",
          id: "57kIMCLPgkzQlXjblX7XXP",
          name: "Phosphorescent",
          type: "artist",
          uri: "spotify:artist:57kIMCLPgkzQlXjblX7XXP"
        }
      ],
      external_urls: {
        spotify: "https://open.spotify.com/album/0TticAuqNj26EhgT2Em3qD"
      },
      href: "https://api.spotify.com/v1/albums/0TticAuqNj26EhgT2Em3qD",
      id: "0TticAuqNj26EhgT2Em3qD",
      images: [
        {
          height: 640,
          url: "https://i.scdn.co/image/ccd62a1aa5c66d581622a1be2c86cfd0299a3b45",
          width: 640
        },
        {
          height: 300,
          url: "https://i.scdn.co/image/cc9f4c667dc6fa65839c121262a7e7ba5dd49547",
          width: 300
        },
        {
          height: 64,
          url: "https://i.scdn.co/image/1ea5645d08a452f50c9720aad99d4ea6e8cdd8b6",
          width: 64
        }
      ],
      name: "Aw Come Aw Wry",
      type: "album",
      uri: "spotify:album:0TticAuqNj26EhgT2Em3qD"
    },
    track_number: 3
  }
];

export const albums = [
  {
    id: "1bypUQ8qclz4N9ocZgAmPj",
    name: "Aw Man Oh Dear",
    image: "https://i.scdn.co/image/172a34d390848ba1615374fa4f77c304de01a919",
    artists: {
      external_urls: {
        spotify: "https://open.spotify.com/artist/77PtLt3o5UjeAT38mVlGaQ"
      },
      href: "https://api.spotify.com/v1/artists/77PtLt3o5UjeAT38mVlGaQ",
      id: "77PtLt3o5UjeAT38mVlGaQ",
      name: "Aw Man Oh Dear",
      type: "artist",
      uri: "spotify:artist:77PtLt3o5UjeAT38mVlGaQ"
    },
    url_songs: "https://api.spotify.com/v1/albums/1bypUQ8qclz4N9ocZgAmPj",
    type: "album"
  }
];
