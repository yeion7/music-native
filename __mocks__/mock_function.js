export const Mock = jest.fn();
