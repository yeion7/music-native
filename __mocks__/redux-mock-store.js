import configureMockStore from "redux-mock-store";

const middlewares = [];
const mockStore = configureMockStore(middlewares);

export default mockStore;
