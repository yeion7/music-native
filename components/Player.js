/* @flow weak */

import React from "react";
import { View, Dimensions } from "react-native";
import { Button, Icon, Spinner } from "native-base";

import PlayList from "./PlayList";
import Controls from "./Controls";
import Info from "./Info";

const { width, height } = Dimensions.get("window");
import styled from "styled-components/native";

const Player = ({
  onPlayPause,
  onNext,
  onBack,
  isLoading,
  isPlaying,
  currentSong,
  playbackInstancePosition,
  playbackInstanceDuration,
  expanded,
  onExpand,
  playList,
  onPressSong
}) => (
  <Container style={{ height: expanded ? height : height * 0.25 }}>
    {playList.length > 0
      ? <Button full black onPress={onExpand}>
          {expanded ? <Icon name="arrow-down" /> : <Icon name="arrow-up" />}
        </Button>
      : null}

    {isLoading
      ? <Spinner color="black" />
      : <Controls
          onPlayPause={onPlayPause}
          onNext={onNext}
          onBack={onBack}
          isPlaying={isPlaying}
        />}

    <Info
      position={playbackInstancePosition}
      duration={playbackInstanceDuration}
      song={currentSong}
    />

    {expanded && <PlayList playList={playList} onPressSong={onPressSong} />}
  </Container>
);

const Container = styled.View`
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  background: white;
  overflow: visible;
  max-height: ${height}
  width: ${width}
  align-items: center;
  justify-content: flex-start;
  flex: 1;
`;

export default Player;
