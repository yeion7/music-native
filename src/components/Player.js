/* @flow weak */

import React from "react";
import { View, Dimensions } from "react-native";
import { Button, Icon, Spinner } from "native-base";
import { connect } from "react-redux";

import PlayList from "./PlayList";
import Controls from "./Controls";
import Info from "./Info";

import {
  expandPlayer,
  handlePlayPause,
  handleForward,
  handleBack,
  changePlayerStatus
} from "../actions/player";

const { width, height } = Dimensions.get("window");
import styled from "styled-components/native";

const Player = ({
  onPlayPause,
  onNext,
  onBack,
  onExpand,
  isLoading,
  isPlaying,
  song,
  position,
  duration,
  expand,
  playlist,
  index,
  showPlayer
}) => (
  <View>
    {showPlayer &&
      <Container
        style={{
          height: expand ? height : height * 0.27,
          justifyContent: expand ? "flex-start" : "center"
        }}
      >
        {playlist.length > 0
          ? <Button full black onPress={onExpand}>
              {expand ? <Icon name="arrow-down" /> : <Icon name="arrow-up" />}
            </Button>
          : null}

        {isLoading
          ? <Spinner color="black" />
          : <Controls
              onPlayPause={onPlayPause}
              onNext={onNext}
              onBack={onBack}
              isPlaying={isPlaying}
              index={index}
            />}

        <Info position={position} duration={duration} song={song} />

        {expand && <PlayList playList={playlist} />}
      </Container>}
  </View>
);

const Container = styled.View`
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  background: white;
  overflow: visible;
  max-height: ${height}
  width: ${width}
  align-items: center;
  flex: 1;
`;

const mapStateToProps = ({
  player: {
    status: { showPlayer, isLoading, isPlaying, position, duration, expand },
    song,
    playlist
  }
}) => ({
  showPlayer,
  isLoading,
  isPlaying,
  song,
  position,
  duration,
  expand,
  playlist,
  index: song.track_number
});

export default connect(mapStateToProps)(Player);
