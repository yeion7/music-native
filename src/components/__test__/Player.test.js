import React from "react";
import "react-native";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";

import { Mock } from "../../../__mocks__/mock_function";
import { song, playlist, status } from "../../../__mocks__/mock_data";
import mockStore from "../../../__mocks__/redux-mock-store";

const store = mockStore({
  player: {
    song,
    playlist,
    status
  }
});

import Player from "../Player";

describe("renders test ", () => {
  it("render Player component", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Player
            onPlayPause={Mock}
            onNext={Mock}
            onBack={Mock}
            onExpand={Mock}
          />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render Player component expanded", () => {
    const tree = renderer
      .create(
        <Provider store={store}>

          <Player
            onPlayPause={Mock}
            onNext={Mock}
            onBack={Mock}
            onExpand={Mock}
          />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render Player component without playlist", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Player
            onPlayPause={Mock}
            onNext={Mock}
            onBack={Mock}
            onExpand={Mock}
          />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
