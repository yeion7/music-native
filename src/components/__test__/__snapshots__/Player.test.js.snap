// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders test  render Player component 1`] = `<View />`;

exports[`renders test  render Player component expanded 1`] = `<View />`;

exports[`renders test  render Player component without playlist 1`] = `<View />`;
